apps/server/src/classes/data-provider/DataProvider.ts

```diff
@@ -32,6 +33,22 @@ export class DataProvider {
     return data.project;
   }
 
+  static async setCustomField(newData: CustomFieldDefinitions) {
+    data.customFields = { ...data.customFields, ...newData };
+    await this.persist();
+    return data.customFields;
+  }
+
+  static async overrideCustomField(newData: CustomFieldDefinitions) {
+    data.customFields = { ...newData };
+    await this.persist();
+    return data.customFields;
+  }
+
+  static getCustomField() {
+    return data.customFields;
+  }
```

Comment:
I think the dataprovider should be pretty dumb.
Meaning we should have a single setter function as we do with the other fields

Also: do you mind typing the return values? it will help us when we go intro strict mode